src/daal4py.cpp

```diff
@@ -678,3 +678,13 @@ void c_train_test_split(data_or_file & orig, data_or_file & train, data_or_file
 #else
 #endif
 }
+
+void c_generate_shuffled_indices(data_or_file & idx, data_or_file & random_state)
```

Comment:
don't you want to merge into function `c_train_test_split`?

Reply:
In some cases optimizations of `c_generate_shuffled_indices` or `c_train_test_split` may not be used. So, to use at least one of them, functions should be separate.